filament/backend/src/metal/MetalDriver.mm

```diff
@@ -1267,29 +1267,29 @@
     // Enumerate all the sampler buffers for the program and check which textures and samplers need
     // to be bound.
 
-    id<MTLTexture> texturesToBind[SAMPLER_BINDING_COUNT] = {};
-    id<MTLSamplerState> samplersToBind[SAMPLER_BINDING_COUNT] = {};
+    id<MTLTexture> texturesToBindVertexShader[backend::MAX_VERTEX_SAMPLER_COUNT] = {};
+    id<MTLSamplerState> samplersToBindVertexShader[backend::MAX_VERTEX_SAMPLER_COUNT] = {};
 
-    enumerateSamplerGroups(program, [this, &texturesToBind, &samplersToBind](
-            const SamplerGroup::Sampler* sampler,
-            uint8_t binding) {
+    enumerateSamplerGroups(program, ShaderType::VERTEX,
+            [this, &texturesToBindVertexShader, &samplersToBindVertexShader](
+                    const SamplerGroup::Sampler* sampler, uint8_t binding) {
         // We currently only support a max of SAMPLER_BINDING_COUNT samplers. Ignore any additional
         // samplers that may be bound.
-        if (binding >= SAMPLER_BINDING_COUNT) {
+        if (binding >= backend::MAX_VERTEX_SAMPLER_COUNT) {
             return;
         }
         const auto metalTexture = handle_const_cast<MetalTexture>(sampler->t);
-        texturesToBind[binding] = metalTexture->swizzledTextureView ? metalTexture->swizzledTextureView
-                                                                    : metalTexture->texture;
+        texturesToBindVertexShader[binding] = metalTexture->swizzledTextureView ? metalTexture->swizzledTextureView
+                                                                                : metalTexture->texture;
 
         if (metalTexture->externalImage.isValid()) {
-            texturesToBind[binding] = metalTexture->externalImage.getMetalTextureForDraw();
+            texturesToBindVertexShader[binding] = metalTexture->externalImage.getMetalTextureForDraw();
         }
 
-        if (!texturesToBind[binding]) {
+        if (!texturesToBindVertexShader[binding]) {
```

Comment:
Might be useful to mention the stage (vertex/fragment) in this warning message (same for fragment lambda).

Reply:
Yes it's helpful. I'm done.